webapp/data_structures/RELATIONALOP.js

```diff
@@ -26,86 +25,83 @@ function RELATIONALOP() {
     }
     RELATIONALOP.prototype.get = function RELATIONALOP() {
 
-        var options={
-        rule:["Operator:==(0),~=(1),<(2),<=(3),>(4),>=(5)",this.rule],
-	    zcr:["Use zero crossing (no: 0), (yes: 1)" ,this.zcr],
-	    Datatype:["Datatype (1=double 3=int32 ...)" ,this.Datatype],
+        var options = {
+        rule:["Operator: ==(0), ~=(1), <(2), <=(3), >(4), >=(5)", this.rule],
+	    zcr:["Use zero crossing (no: 0), (yes: 1)" , this.zcr],
+	    Datatype:["Datatype (1=double 3=int32 ...)" , this.Datatype],
         }
         return options
     }
     RELATIONALOP.prototype.set = function RELATIONALOP() {
 
-    this.rule = parseInt((arguments[0]["rule"]))
-	this.zcr = parseInt((arguments[0]["zcr"]))
-	this.Datatype = parseInt((arguments[0]["Datatype"]))
-
-	if(this.zcr< 0 || this.zcr >1)
-	{
-                alert("incorrect operator");
-		RELATIONALOP.get();
-	}
+        this.rule = parseInt((arguments[0]["rule"]));
+	    this.zcr = parseInt((arguments[0]["zcr"]));
+	    this.Datatype = parseInt((arguments[0]["Datatype"]));
+	    var zcr1 = 0;
+	    if(this.zcr != 0){
+            zcr1 = 1;
+	    }
+	    if(this.rule < 0 || this.rule > 5){
+            alert("Incorrect operator "+this.rule.toString()+" ; must be 0 to 5.");
+            throw "incorrect";
+        }
 		if (this.rule == 0 ){
```

Comment:
```suggestion
                var label;
		switch (this.rule) {
		    case 0: label = "=="; break;
		    case 1: label = "~=" break;
```
use switch here.

do not use this with label.